heapify.mjs

```diff
@@ -153,10 +153,25 @@ export default class Heapify {
     }
 
     toString() {
+        if (this.length - ROOT_INDEX < 0 ) {
+            return '(empty queue)'
+        }
+
         let result = Array(this.length - ROOT_INDEX);
         for (let i = 0; i < this.length; i++) {
             result[i] = this.priorities[i + ROOT_INDEX];
         }
         return `[${result.join(" ")}]`;
+    
+    }
+
+    get [Symbol.toStringTag]() {
+        return 'Heapify'
+    }
+
+    * [Symbol.iterator] () {
+        for (let i = 0; i < this.length; i++) {
+            yield this.priorities[i + ROOT_INDEX];
```

Comment:
I have mixed feelings about this. Normally a priority queue would output keys by default. That's what `peek()` and `pop()` do. So, to make it consistent with the existing API, I think this should iterate through the keys instead. Makes sense?

Reply:
Makes sense, but I am curious about the why in your `.toString()` you use priorities. Also should we iterate in order of priority?

What if the default iterator yields tuples (key, priority)?

Example:

```javascript
const queue = new Heapify();
queue.push(5, 35);
queue.push(3, 30);
queue.push(1, 10);
queue.push(2, 20);
queue.push(4, 40);

console.log([...queue]) // [ [ 1, 10 ], [ 2, 20 ], [ 3, 30 ], [ 5, 35 ], [ 4, 40 ] ]
```

We could also add a `.keysIter` and a `.prioritiesIter` iterator to get only keys and priorities where needed.

Examples:

```javascript
const queue = new Heapify();
queue.push(5, 35);
queue.push(3, 30);
queue.push(1, 10);
queue.push(2, 20);
queue.push(4, 40);

console.log([...queue.keysIter()]); // [ 1, 2, 3, 5, 4 ]
```

And

```javascript
const queue = new Heapify();
queue.push(5, 35);
queue.push(3, 30);
queue.push(1, 10);
queue.push(2, 20);
queue.push(4, 40);

console.log([...queue.prioritiesIter()]); // [ 10, 20, 30, 35, 40 ] 
```

Reply:
> Makes sense, but I am curious about the why in your .toString() you use priorities.

Yeah, good point; it's wrong. I probably used it during debugging and forgot to remove it. I will fix it in a later PR.

Nice, I liked yielding tuples. Let's keep it.

About `keysIter()` and `prioritiesIter()`, not bad, but what about renaming them to just `keys()` and `priorities()`? You can go ahead and rename the internal properties `keys` and `priorities` to `_keys` and `_priorities`; the interface has precedence when it comes to using good names.

I am curious as to what would be valid use cases for iterating through all key or priorities. Are you adding the ability to iterate due to a specific need you have?

Reply:
Changes applied.

Regarding the methods, I think they can be useful for general introspection (I imagine debugging scenarios or to build interfaces that allows to visualize the state of the queue).